File: src/main/java/io/dddrive/property/model/EnumSetProperty.java
package io.dddrive.property.model;

import java.util.Collection;
import java.util.Set;

import io.dddrive.enums.model.Enumerated;

public interface EnumSetProperty<E extends Enumerated> extends CollectionProperty<E> {

	Set<E> getItems();

	boolean hasItem(E item);

	void clearItems();

	void addItem(E item);

	void removeItem(E item);

	void loadEnums(Collection<? extends AggregatePartItem<?>> enums);

}

File: fm-dddrive/src/main/java/io/dddrive/property/model/EnumSetProperty.kt
package io.dddrive.property.model

import io.dddrive.enums.model.Enumerated

/**
 * Property that holds a set of enum values.
 *
 * Implements [Iterable] and [Collection] interfaces so it can be used directly as a set in consumer code.
 */
interface EnumSetProperty<E : Enumerated> :
	Property<E>,
	Collection<E> {

	fun has(item: E): Boolean

	fun clear()

	fun add(item: E)

	fun remove(item: E)

}
